python-checks/src/main/java/org/sonar/python/checks/IdentityComparisonWithNewObjectCheck.java

```diff
@@ -78,16 +83,30 @@ private static boolean checkOperand(Expression operand, IsExpression isExpr, Sub
     Optional<List<Tree>> secondariesOpt = findIssueForOperand(operand);
     secondariesOpt.ifPresent(secondaryLocations -> {
       Token notToken = isExpr.notToken();
-      PreciseIssue preciseIssue = notToken == null ?
-        ctx.addIssue(isExpr.operator(), MESSAGE_IS) :
-        ctx.addIssue(isExpr.operator(), notToken, MESSAGE_IS_NOT);
+      var issue = notToken == null ?
```

Comment:
We seem to be testing twice for the presence of the `not` token (once in the issue creation and one in the quick fix creation), maybe we could only do it once and create the issue and quick fix at the same time? (Basically like in the PR for SONARPY-1084).

Reply:
done